### PR TITLE
Add wrapper for ServerMutex

### PR DESCRIPTION
**Pull request by thepianoboy**
I created `ServerMutex` to wrap the relevant parts of the OnlineMultiplayer object. The major player here is the `lock()` method, which blocks until the game data is locked using an exponential back-off as recommended by the Dropbox API.

### CODE REVIEW

File: core/src/com/unciv/ui/worldscreen/mainmenu/DropBox.kt [hunk 1]
```diff
@@ -219,4 +221,77 @@ class LockFile {
...
+            }
+
+            // Consider NOT sleeping here, instead perhaps delay or spin+yield
+			Thread.sleep(delay)
+
+            tries++
+
+            // Retry the lock
+            locked = tryLock()
+		}
```
Comment (GGGuenni) [inline core/src/com/unciv/ui/worldscreen/mainmenu/DropBox.kt]: The indentation looks kinda wonky and should probably be adjusted for readability. Seems like there are sometimes tabs, sometimes two spaces, and sometimes more? Four spaces is what the rest of the code is going with (I think four spaces is also the default Kotlin setting in Android Studio)

File: core/src/com/unciv/ui/worldscreen/mainmenu/DropBox.kt [hunk 2]
```diff
@@ -219,4 +221,77 @@ class LockFile {
...
+
+    /**
+     * Try to obtain the server lock ONCE
+     * @see OnlineMultiplayer.tryLockGame
+     * @see lock
+     * @see unlock
+     * @return true if lock is acquired
+     */
+	fun tryLock(): Boolean {
+		locked = OnlineMultiplayer().tryLockGame(gameInfo.asPreview())
```
Comment (GGGuenni) [inline core/src/com/unciv/ui/worldscreen/mainmenu/DropBox.kt]: Should we move the OnlineMultiplayer calls into the ServerMutex? I mean its called only once by ServerMutex and having two ways to lock is probably just confusing so we could move the logic from OnlineMultiplayer to ServerMutex

File: core/src/com/unciv/ui/worldscreen/mainmenu/DropBox.kt [hunk 3]
```diff
@@ -219,4 +221,77 @@ class LockFile {
...
+		}
+	}
+
+    /**
+     * Release a server lock acquired by tryLock or lock
+     * @see tryLock
+     * @see lock
+     */
+	fun unlock() {
+		OnlineMultiplayer().tryReleaseLockForGame(gameInfo.asPreview())
```
Comment (GGGuenni) [inline core/src/com/unciv/ui/worldscreen/mainmenu/DropBox.kt]: Same as above

File: core/src/com/unciv/ui/worldscreen/mainmenu/DropBox.kt (hunk 1 above)
Comment (thepianoboy) [reply core/src/com/unciv/ui/worldscreen/mainmenu/DropBox.kt]: Oops... Sometimes I make tweaks before a commit using vim; it probably threw in a few tabs on me. Android Studio renders those correctly, so I never noticed. I'll fix them.

File: core/src/com/unciv/ui/worldscreen/mainmenu/DropBox.kt (hunk 2 above)
Comment (thepianoboy) [reply core/src/com/unciv/ui/worldscreen/mainmenu/DropBox.kt]: I considered that as well. I wasn't sure if you had a particular design in mind when you created those calls, but if they can be moved that's fine too. I'll take care of it.

**Comment by thepianoboy** [comment]
Made the requested changes. Anything else?

**Comment by GGGuenni** [comment]
Looks good I will make some small adjustments regarding the previews myself :+1:

**Pull request merged**